utils/ansible.py

```diff
@@ -0,0 +1,281 @@
+from distutils.sysconfig import get_python_lib
+import tempfile
+from os import listdir, mkdir, path
+from shutil import copy, copyfile, rmtree
+from subprocess import check_output, CalledProcessError, STDOUT
+
+from fauxfactory import gen_alphanumeric
+from utils import conf
+from utils.providers import get_crud
+
+from git import Repo
+from yaml import load, dump
+
+
+yml_path = path.join(path.dirname(__file__) + "/manageiq_ansible_module/")
+library_path = get_python_lib()
+yml_templates_path = path.dirname(__file__) + "/ansible_conf/"
+providers_basic_script = "providers_basic_script.yml"
+users_basic_script = "users_basic_script.yml"
+basic_script = "basic_script.yml"
+basic_script_bk = "basic_script_bk.yml"
+yml = ".yml"
+random_token = str(gen_alphanumeric(906))
+random_miq_user = str(gen_alphanumeric(8))
+local_git_repo = "manageiq_ansible_module"
+pulled_repo_library_path = local_git_repo + "/library"
+remote_git_repo_url = "git://github.com/dkorn/manageiq-ansible-module.git"
+
+
+def create_tmp_directory():
+    global lib_path
+    lib_path = tempfile.mkdtemp()
+    lib_sub_path = 'ansible_conf'
+    lib_sub_path_library = path.join(lib_sub_path, 'library')
+    folders = ['ansible_conf', lib_sub_path_library]
+    for folder in folders:
+        mkdir(path.join((lib_path), folder))
+    global library_path_to_copy_to
+    global basic_yml_path
+    library_path_to_copy_to = path.join(lib_path, lib_sub_path_library)
+    basic_yml_path = path.join(lib_path, lib_sub_path)
+
+
+def fetch_miq_ansible_module():
+    if path.isdir(local_git_repo):
+        rmtree(local_git_repo)
+    mkdir(local_git_repo)
+    if path.isdir(library_path_to_copy_to):
+        rmtree(library_path_to_copy_to)
+    mkdir(library_path_to_copy_to)
+    Repo.clone_from(remote_git_repo_url, local_git_repo)
+    src_files = listdir(pulled_repo_library_path)
```

Comment:
OPTIONAL: I guess you can just copy the folder by copytree and rename it instead of all these operations.